src/megaapi.cpp

```diff
@@ -3937,576 +3937,574 @@ char *MegaApi::getMimeType(const char *extension)
-        (*mimeMap)["wmv"]="video/x-ms-wmv";
-        (*mimeMap)["wmx"]="video/x-ms-wmx";
-        (*mimeMap)["wmz"]="application/x-ms-wmz";
-        (*mimeMap)["wpl"]="application/vnd.ms-wpl";
-        (*mimeMap)["wps"]="application/vnd.ms-works";
-        (*mimeMap)["wri"]="application/x-mswrite";
-        (*mimeMap)["wrl"]="x-world/x-vrml";
-        (*mimeMap)["wrz"]="x-world/x-vrml";
-        (*mimeMap)["wsc"]="text/scriptlet";
-        (*mimeMap)["wsdl"]="text/xml";
-        (*mimeMap)["wvx"]="video/x-ms-wvx";
-        (*mimeMap)["x"]="application/directx";
-        (*mimeMap)["xaf"]="x-world/x-vrml";
-        (*mimeMap)["xaml"]="application/xaml+xml";
-        (*mimeMap)["xap"]="application/x-silverlight-app";
-        (*mimeMap)["xbap"]="application/x-ms-xbap";
-        (*mimeMap)["xbm"]="image/x-xbitmap";
-        (*mimeMap)["xdr"]="text/plain";
-        (*mimeMap)["xht"]="application/xhtml+xml";
-        (*mimeMap)["xhtml"]="application/xhtml+xml";
-        (*mimeMap)["xla"]="application/vnd.ms-excel";
-        (*mimeMap)["xlam"]="application/vnd.ms-excel.addin.macroEnabled.12";
-        (*mimeMap)["xlc"]="application/vnd.ms-excel";
-        (*mimeMap)["xld"]="application/vnd.ms-excel";
-        (*mimeMap)["xlk"]="application/vnd.ms-excel";
-        (*mimeMap)["xll"]="application/vnd.ms-excel";
-        (*mimeMap)["xlm"]="application/vnd.ms-excel";
-        (*mimeMap)["xls"]="application/vnd.ms-excel";
-        (*mimeMap)["xlsb"]="application/vnd.ms-excel.sheet.binary.macroEnabled.12";
-        (*mimeMap)["xlsm"]="application/vnd.ms-excel.sheet.macroEnabled.12";
-        (*mimeMap)["xlsx"]="application/vnd.openxmlformats-officedocument.spreadsheetml.sheet";
-        (*mimeMap)["xlt"]="application/vnd.ms-excel";
-        (*mimeMap)["xltm"]="application/vnd.ms-excel.template.macroEnabled.12";
-        (*mimeMap)["xltx"]="application/vnd.openxmlformats-officedocument.spreadsheetml.template";
-        (*mimeMap)["xlw"]="application/vnd.ms-excel";
-        (*mimeMap)["xml"]="text/xml";
-        (*mimeMap)["xmta"]="application/xml";
-        (*mimeMap)["xof"]="x-world/x-vrml";
-        (*mimeMap)["XOML"]="text/plain";
-        (*mimeMap)["xpm"]="image/x-xpixmap";
-        (*mimeMap)["xps"]="application/vnd.ms-xpsdocument";
-        (*mimeMap)["xrm-ms"]="text/xml";
-        (*mimeMap)["xsc"]="application/xml";
-        (*mimeMap)["xsd"]="text/xml";
-        (*mimeMap)["xsf"]="text/xml";
-        (*mimeMap)["xsl"]="text/xml";
-        (*mimeMap)["xslt"]="text/xml";
-        (*mimeMap)["xsn"]="application/octet-stream";
-        (*mimeMap)["xss"]="application/xml";
-        (*mimeMap)["xtp"]="application/octet-stream";
-        (*mimeMap)["xwd"]="image/x-xwindowdump";
-        (*mimeMap)["z"]="application/x-compress";
-        (*mimeMap)["zip"]="application/x-zip-compressed";
-    }
+    static map<string, string> mimeMap{
+        // from c++11, this sort of static local initialization is one-time and thread safe
```

Comment:
cool, this comment is meant to be specific to the usage here, so our peers who see it will know it's ok, and where to look it up if needed (like me :-) )